virttest/utils_net.py

```diff
@@ -3026,7 +3026,12 @@ def get_host_ip_address(params):
         raise NetError("Error while retrieving IP address for host.")
     logging.warning("No IP address of host was provided, using IP address"
                     " on %s interface", str(net_dev))
-    host_ip = get_ip_address_by_interface(net_dev)
+    # host_ip = get_ip_address_by_interface(net_dev)
+    ips = get_net_if_and_addrs()
+    ip_v = params.get('listening_addr')
+    if not ip_v:
+        ip_v = 'ipv4'
+    host_ip = ips[net_dev][ip_v][0]
```

Comment:
Very-very dangerous code.
There is no any guaranty that ips holds 'net_dev' and it holds 'ip_v' and it has at least one address.